src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { Hex } from 'viem';
+import { faker } from '@faker-js/faker';
+import { ProxyFactoryDecoder } from '@/domain/relay/contracts/proxy-factory-decoder.helper';
+import { createProxyWithNonceEncoder } from '@/domain/relay/contracts/__tests__/proxy-factory-encoder.builder';
+
+describe('ProxyFactoryDecoder', () => {
+  let target: ProxyFactoryDecoder;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    target = new ProxyFactoryDecoder();
+  });
+
+  it('decodes a createProxyWithNonce function call correctly', () => {
+    const createProxyWithNonce = createProxyWithNonceEncoder();
+    const args = createProxyWithNonce.build();
```

Comment:
I would move this to the "assert" section and call it `expectedArgs` (as it's only used in the assertion). What do you think?

Reply:
Makes sense to me - changed in 192ff518bad0aff21b70ad12ca180db3b701020b.